src/targets/gpu/mlir.cpp

```diff
@@ -580,12 +599,93 @@ struct mlir_program
+        std::string tuning_db_path = string_value_of(MIGRAPHX_MLIR_TUNING_DB{});
+        if(!tuning_db_path.empty())
+        {
+            std::ifstream tuning_db_tsv(tuning_db_path);
+            if(tuning_db_tsv)
+            {
+                std::string arch, prob, perf, key;
+                std::getline(tuning_db_tsv, arch);
+                while(std::getline(tuning_db_tsv, arch, '\t'))
+                {
+                    std::getline(tuning_db_tsv, prob, '\t');
+                    std::getline(tuning_db_tsv, perf);
+                    key             = arch + "\t" + prob;
+                    char* key_cstr  = strdup(key.c_str());
+                    char* perf_cstr = strdup(perf.c_str());
+                    mlirRockTuningUpdateTable(tuning_table, key_cstr, perf_cstr, 1.0);
+                }
+                return;
+            }
+        }
+        std::cerr << "WARNING: MLIR tuning db not found. Please set MIGRAPHX_MLIR_TUNING_DB for "
+                     "optimal performance."
+                  << std::endl;
+    }
+
+    // This function appends to tuning cfg file that could be
+    // used with rocMLIR tuning scripts.
+    void dump_tuning_cfg(const char* probConfig)
+    {
+        std::string tuning_cfg_path = string_value_of(MIGRAPHX_MLIR_TUNING_CFG{});
+        if(!tuning_cfg_path.empty())
+        {
+            std::stringstream key(probConfig);
+            std::string prob;
+            int token_count = 0;
+            for(std::string s; getline(key, s, '\t');)
+            {
+                // first token is arch
+                // second one is the problem cfg
+                if(token_count == 1)
+                {
+                    prob = s;
+                }
+                if(token_count > 1)
+                {
+                    break;
+                }
+                token_count++;
+            }
+
+            if(prob.substr(0, 4) == std::string("conv"))
+            {
+                tuning_cfg_path = tuning_cfg_path + ".conv";
+            }
+            else
+            {
+                tuning_cfg_path = tuning_cfg_path + ".gemm";
+            }
+            std::ofstream tuning_cfg(tuning_cfg_path, std::ios::app);
+            tuning_cfg << prob << std::endl;
+        }
+    }
+
+    bool get_module_tuned()
+    {
+        if(!mlirRockTuningSetFromTable(tuning_table, mmodule.get()))
+        {
+            const char* probConfig = mlirRockTuningGetKey(tuning_table, mmodule.get());
+            std::stringstream key(probConfig);
+            std::cerr << "fails to set param on" << probConfig << std::endl;
+            dump_tuning_cfg(probConfig);
+            return false;
+        }
+        return true;
+    }
+
     mlir_context ctx;
     MlirLocation location;
     mlir_module mmodule;
     problem_params pp;
     std::deque<std::string> strings{};
     std::string target_arch;
+    MlirRockTuningTable tuning_table;
```

Comment:
This should just be stored in the `get_module_tuned`. I dont see why it needs to be a member variable.

Reply:
Actually this should be a static variable so we only read the db file once and not everytime we do a mlir compile.